### ISSUE TITLE
[BUG] RSO missing cpid in userinfo

### ISSUE THREAD

**Issue by whyando** [RiotGames/developer-relations#992]
cpid is missing from the response from `https://auth.riotgames.com/userinfo`
cpid scope IS specified.

normally `https://auth.riotgames.com/userinfo` returns  { sub, cpid, jti } but just getting { sub }

* Frequency:
Not every account is affected. It may only be for new accounts. I am unsure when this bug started occurring.

**Comment by colin-j-rm** [comment]
Also running into this issue. I can replicate it consistently, only happens when account is new and you need to first click "Authorize", then no matter how many times you retry, /userinfo doesn't return a cpid and only returns { sub }. 

(if you restart entire sign in flow then it finally works, but this is not ideal for users forcing them to retry signing in)

**Comment by OllieJennings** [comment]
I have also started to notice this happening. @tisbells any knowledge of this, we are seeing multiple signups with no `cpid` coming back

**Comment by DarkIntaqt** [comment]
I also had these issues a lot recently…
Tisbells isn’t working for DevRel anymore, so this ping is unfortunately not helping

**Comment by OllieJennings** [comment]
Oh no, @DarkIntaqt is there any rioters that we know of who we can tag?

I feel like this has been getting worse.

**Comment by DarkIntaqt** [comment]
afaik they are a heavily short staffed atm, and while I agree with you that this has a pretty high priority, I don't know whether there is anyone who is actively in charge of / monitoring the Github.

**Comment by OllieJennings** [comment]
I think you are right, the only developer l have seen who has replied on some github issues is:  @lucasMesquitaBorges, any chance we could get some eyes on this.

I've also submitted a ticket on this: 108409707

**Comment by lucasMesquitaBorges** [comment]
Hi! That's correct. Thanks for the ping in this thread.
I will check this issue.

**Comment by OllieJennings** [comment]
Hey @lucasMesquitaBorges any update on this? We are now seeing around 1 in 3 requests without a `cpid` field. 

For now we have resorted to just spamming the summoner endpoint for each region to find out which region they might belong to, which is not ideal but gives up an alternative for now.

**Comment by lucasMesquitaBorges** [comment]
The `cpid` object from `/userinfo` endpoint will only return if the player has played lol (that's when a shard is bound to the account).
For the given rate you are reporting this issue, I'm assuming this `cpid` object is missing for players with a bound shard.

To continue triaging this issue we will need to get some sensitive information.

@OllieJennings since you provided a ticket ID, we will contact you via zendesk to get more information and find the root cause.

**Comment by OllieJennings** [comment]
@lucasMesquitaBorges yep exactly that, we are finding that we find the "shard" by looping through all regions and requests their summoner based on their `puuid`.

More that happy to give all required details when needed, will keep an eye out for the ticket response.

**Update**: have responded back to the ticket with the relevant info

**Comment by lucasMesquitaBorges** [comment]
Hi all! Can you please detail what scopes you're including in the redirect url for user authorization?
(URL redirect to `https://auth.riotgames.com/authorize`)

**Comment by KarineValenca** [comment]
Hey @lucasMesquitaBorges 

I'm facing the same issue on my application. The scopes I'm using are: `openid` and `cpid`.

**Comment by lucasMesquitaBorges** [comment]
Please add `offline_access` to your scopes and check if the error still happens!

**Comment by lucasMesquitaBorges** [comment]
@whyando @colin-j-rm @DarkIntaqt @OllieJennings @KarineValenca 

> Please add `offline_access` to your scopes and check if the error still happens!

Can you check if this solves your issue?

**Comment by DarkIntaqt** [comment]
> Can you check if this solves your issue?

For some reason that I can't explain it seems to work without offline_access as well suddenly. Let's wait for someone else to respond...

**Comment by OllieJennings** [comment]
@lucasMesquitaBorges Will try it out this week!

**Comment by KarineValenca** [comment]
> > Can you check if this solves your issue?
> 
> For some reason that I can't explain it seems to work without offline_access as well suddenly. Let's wait for someone else to respond...

@DarkIntaqt In my case, it happened only during the first authentication for the user, so if you are doing a test with a user who was already authenticated in our app, this could explain the sudden fix.

Thanks @lucasMesquitaBorges, I'll test it out this week and update it here.

**Comment by lucasMesquitaBorges** [comment]
Thanks to all of you for reporting this issue. I'm pretty sure it was already fixed. Feel free to ping in case you need any help!

**Comment by KarineValenca** [comment]
hey @lucasMesquitaBorges sorry for the delay. I made the change you proposed, adding the offline_access, but it didn't work. Here's the behavior I detected: When it's the first time the user authenticates with my client, the CPID is not returned. If the user tries the authentication process again, the CPID is returned correctly so no bug. I asked 2 different people who never connected their riot account with my app to test it, and in both cases, the same behavior happened.
